src/translator/sentence_ranges.h

```diff
@@ -3,50 +3,120 @@
 
 #include "data/types.h"
 #include <cassert>
+#include <utility>
 #include <vector>
 
 namespace marian {
 namespace bergamot {
 
-class SentenceRanges {
-  // SentenceRanges stores string_views into a source text, with additional
-  // annotations to mark sentence boundaries.
-  //
-  // Given the availability annotations, this container provides capabilty to
-  // add sentences, and access individual sentences.
+/// ByteRange stores indices for begin and end (inclusive) in a string. Can be
+/// used to represent a sentence, word.
+struct ByteRange {
+  size_t begin_byte_offset;
+  size_t end_byte_offset;
+  const size_t size() const { return end_byte_offset - begin_byte_offset + 1; }
```

Comment:
@abhi-agg Neater I think is `end_byte_offset` not inclusive. Can I make the switch? You mentioned only the struct not what the components mean.

Reply:
C++ standard usage is [begin, end) with half-open intervals.  Therefore this should be end_byte_offset - begin_byte_offset.

Reply:
Fine with it 👍